File: e2e/java/src/main/java/io/trzcinski/test/store/StoreClient.java
package io.trzcinski.test.store;

import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Map;
import org.springframework.lang.Nullable;

import io.trzcinski.test.commons.dto.Order;

interface StoreClient {

    @GetMapping("/store/inventory")
    Map<String, Object> getInventory(
    );

    @PostMapping("/store/order")
    Order placeOrder(
        @RequestBody() Order payload
    );

    @GetMapping("/store/order/{orderId}")
    Order getOrderById(
        @RequestParam("orderId") Integer orderId
    );

    @DeleteMapping("/store/order/{orderId}")
    void deleteOrder(
        @RequestParam("orderId") Integer orderId
    );
}

File: test-samples/kotlin-retrofit/src/main/kotlin/io/trzcinski/test/store/StoreClient.kt
package out.io.trzcinski.test.store

import retrofit2.http.*
import retrofit2.Call

import out.io.trzcinski.test.commons.dto.Order

interface StoreClient {

    @GET("/store/inventory")
    fun getInventory(
    ): Call<Map<Object, Object>>

    @POST("/store/order")
    fun placeOrder(
    ): Call<Order>

    @GET("/store/order/{orderId}")
    fun getOrderById(
        @Path("orderId") orderId: Int
    ): Call<Order>

    @DELETE("/store/order/{orderId}")
    fun deleteOrder(
        @Path("orderId") orderId: Int
    ): Call<Void>
}